### ISSUE TITLE
no-empty-character-class triggered incorrectly when 'u' flag is present

### ISSUE THREAD

**Issue by bathos** [eslint/eslint#3679]
The presence of the 'u' flag is triggering the no-empty-character-class rule.

Triggers:
    const pattern = /hello/u;

Does not trigger:
    const pattern = /hello/;

I'm not sure how this can be because the source at https://github.com/eslint/eslint/blob/master/lib/rules/no-empty-character-class.js suggests this pattern should be fine. I checked my version with `eslint --version` and it appears it should be current with that file (1.31). Can anyone replicate this?

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by ilyavolodin** [comment]
I can't reproduce this in online demo. Following code is lint-free for me:
```js
/* eslint no-empty-character-class:2 */
const pattern = /hello/u;
```

**Comment by bathos** [comment]
Honestly I'm not surprised. I've been puzzling over this for a while and I still haven't figured it out, but it seems like I must have some older version hiding somewhere. I'll close this -- whatever it is, it isn't eslint.

**Comment by bathos** [comment]
I'm an idiot -- I forgot that babel-eslint is in the mix. (You guys must get that all the time.) Turning it off does fix this (even though it breaks future ES stuff), so it may actually be an issue over there.

**Comment by ilyavolodin** [comment]
No problem, happens all the time:-) Yes, looks like an issue with `babel-eslint`, you might want to file a ticket for them.
